lib/pages/main_page.dart

```diff
@@ -1,18 +1,39 @@
 import 'package:flutter/material.dart';
-import 'task_page.dart';
 import '../models/task.dart';
+import '../widgets/task_tile.dart';
+import '../widgets/create_task.dart';
+
+// Список задач
 
 class MainPage extends StatefulWidget {
   @override
   _MainPageState createState() => _MainPageState();
 }
 
 class _MainPageState extends State<MainPage> {
+  TextEditingController _controller = TextEditingController();
   List tasksList = [
-    Task('Задача 1', false, 0, 0),
-    Task('Задача 2', false, 2, 4),
-    Task('Задача 3', false, 1, 3),
+    Task(1, 'Задача 1', false, 0, 0),
+    Task(2, 'Задача 2', false, 2, 4),
+    Task(3, 'Задача 3', false, 1, 3),
   ];
+  List filteredTasksList = [];
+  bool isFiltered = false;
+
+  void _filterTasks() {
+    if (!isFiltered) {
+      if (!tasksList.every((task) => task.isComplete == false)) {
+        setState(() {
+          filteredTasksList =
+              tasksList.where((task) => task.isComplete == false).toList();
```

Comment:
Так будет лаконичнее
```suggestion
              tasksList.where((task) => !task.isComplete).toList();
```